src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1834,7 +1834,7 @@ boolean isSameName(String name) {
         @Override
         boolean isSamePrincipal(final Principal principal) {
             String name = capturedIdentity.getSecurityDomain().getPrincipalDecoder().getName(principal);
-            return isSameName(name);
+            return name != null ? isSameName(name) : false;
```

Comment:
I am not 100% sure about this - would need to be sure of the side effects of returning false rather than reporting an error.